plugin/storage/es/options.go

```diff
@@ -325,3 +333,18 @@ func (opt *Options) Get(namespace string) *config.Configuration {
 func stripWhiteSpace(str string) string {
 	return strings.Replace(str, " ", "", -1)
 }
+
+func initDateLayout(separator string) string {
```

Comment:
I think it is a good idea to set the default separator to "-". But I think configuring `--es.index-date-separator=` as not configuring this should have the same effect, otherwise it may cause some confusion. What do you think?

Reply:
Yup, I see your point. However, IMHO, if I see a flag called `index-date-separator` with a default single char value of `-`, and I needed `yyyyMMdd` index name formatting, my first instinct would be to try `--es.index-date-separator=` or, failing that, I'd try `--es.index-date-separator=''` (but the latter would take those quotes literally as separators...); then I'd re-read the help documentation :)

Would a user who doesn't want to configure this flag still add `--es.index-date-separator=`? If so, what would be the use case?

That's just my two-cents; perhaps other reviewers/maintainers have a different opinion/view? I'm happy to go with the majority. :)

Reply:
> But I think configuring --es.index-date-separator= as not configuring this should have the same effect

I don't think so. If you don't configure it, the standard `-` default applies. Setting it to blank is not the same as not setting.

Reply:
I got it,  I'll modify the code according to this.